04/r/particulate-matter.Rmd

```diff
@@ -0,0 +1,385 @@
+
+- It means the prior doesn't *really* represent our prior knowledge, which defeats the entire conceptual point of this branch of statistics
+- More practically, it means that the data has to do more *work* than it should to constrain the prior. We'll tend to get posteriors that overestimate uncertainty and (if we do a really bad job specifying the prior) may be biased.
+
+So what should we be looking for? From *Gabry et al*,
+
+> What do we need in our priors? This suggests we need *containment*: priors that keep us inside sensible parts of the parameter space.
+
+So our priors should be broader than (and include) our data - but not be so broad that they make impossible predictions. Let's try again with our intercept constrained to be near zero, slope biased toward being a small poisitive number (since we already know the measurements are roughly correlated) and noise confined to near the unit scale:
+
+```{r informative-prior, comment=NA}
+prior_file <- "../../stan/particulate_matter_ols_prior.stan"
+cat(read_file(prior_file))
+```
+
+Compile this stan model.
+```{r stan-compile-priors}
+prior_model <- stan_model(prior_file)
+```
+
+Now let's sample from this model.
+```{r stan-sample-priors}
+prior_samples <- sampling(prior_model, data = prior_data, iter = 300, chains = 1)
+```
+
+```{r prior-extract-samples}
+log_pm_prior <- prior_samples %>% 
+    spread_draws(log_pm_prior[city_id]) %>% 
+    ungroup()
+
+log_pm_prior
+```
+
+Let's take a look at 9 random draws from our estimates. We'll also join in the original data (by `city_id`) so we can compare against.
+```{r prior-log-pm-comparison}
+log_pm_comparison <- log_pm_prior %>% 
+    filter(.draw %in% sample(max(.draw), size = 9L)) %>% 
+    left_join(data %>% select(city_id, log_sat, log_pm25), by = "city_id") %>% 
+    select(.draw, city_id, log_sat, observed = log_pm25, sampled = log_pm_prior)
+
+log_pm_comparison
+```
+
+Let's visualize these results to see if our prior distribution is producing feasible values for `log_pm25`.
+```{r plot-prior-log-pm-comparison}
+log_pm_comparison %>% 
+    gather("type", "log_pm_25", observed:sampled) %>% 
+    ggplot(aes(x = log_sat, y = log_pm_25, color = type)) +
+    geom_point(alpha = 0.3) + 
+    facet_wrap(~ .draw)
+```
+
+## 4. Baseline Model Inference
+Here's Gabry *et al's* code for the basic regression model:
+
+```{r regression-model, comment=NA}
+model_file <- "../../stan/particulate_matter_ols.stan"
+cat(read_file(model_file))
+```
+Compile this stan model.
+```{r stan-compile-regression}
+model <- stan_model(model_file)
+```
+
+Now let's sample from this model.
+```{r stan-sample-regression}
+model_data <- list(N = nrow(data), log_sat = data$log_sat, log_pm = data$log_pm25)
+model_samples <- sampling(model, data = model_data, iter = 1000, chains = 4)
```

Comment:
Slightly different from python version, which has 10,000 iterations. Do we care about that?